graphql_compiler/schema_transformation/rename_schema.py

```diff
@@ -388,8 +602,161 @@ def enter_field_definition(
             new_field_name = self.renamings.get(field_name, field_name)  # Default use original
             if new_field_name == field_name:
                 return IDLE
+            if new_field_name is None:
+                # Suppress the type
+                return REMOVE
             else:  # Make copy of node with the changed name, return the copy
                 field_node_with_new_name = get_copy_of_node_with_new_name(node, new_field_name)
                 return field_node_with_new_name
 
         return IDLE
+
+
+class CascadingSuppressionCheckVisitor(Visitor):
+    def __init__(self, renamings: Mapping[str, Optional[str]], query_type: str) -> None:
+        """Create a visitor to check that suppression does not cause an illegal state.
+
+        Args:
+            renamings: maps original type name to renamed name or None (for type suppression). Any
+                       name not in the dict will be unchanged
+            query_type: name of the query type (e.g. RootSchemaQuery)
+        """
+        self.renamings = renamings
+        self.query_type = query_type
+        self.current_type: Optional[str] = None
+        # Maps a type T to a dict which maps a field F belonging to T to the field's type T'
```

Comment:
I'm not sure I understand this. The outer `str` is the name of a vertex type T. And then `field F belonging to T to the field's type T` maps each of T's fields to T's name? It seems like the values in the inner dict are redundant as described.

Reply:
Consider this part of a schema:
```
type Foo {
  ...
}
type Bar {
  baz: Foo
}
```
Then `F` is `baz`, `T` is `Bar`, and `T'` is Foo. I include `T'` so we can specify precisely which type `T'`'s suppression would cause a cascading error.

Reply:
I see. I was confused because I didn't realize `F` is in effect just a name while `T/T'` are types.

How about this alternative:
```
For type named T, and its field named F whose type has name V, this dict would be {"T": {"F": "V"}}
```
`T` and `T'` being named similar things is somewhat misleading since they don't have to be related types — any type names would work.